staging/src/k8s.io/client-go/tools/cache/reflector_test.go

```diff
@@ -72,7 +74,7 @@ func TestCloseWatchChannelOnError(t *testing.T) {
 			return &v1.PodList{ListMeta: metav1.ListMeta{ResourceVersion: "1"}}, nil
 		},
 	}
-	go r.ListAndWatch(wait.NeverStop)
+	go func() { assert.NoError(t, r.ListAndWatchWithContext(ctx)) }()
```

Comment:
@p0lyn0mial this is correct?

Reply:
Preexisting, this has nothing to do with the watch-list :)

I think the intention of the test is to validate whether the watch will be closed on error. 

Regarding running `ListAndWatchWithContext` in a separate goroutine, I think it should be fine because the ctx will be closed at the end of the test. We will either have a chance to validate the error (which will be nil) or not, depending on whether the runtime preempts the goroutine.